src/character/mod.rs

```diff
@@ -61,14 +68,16 @@ impl Character {
 
     pub fn is_player(&self) -> bool {
         // kind of ugly but does the job
-        self.class.name == "hero"
+        self.class.name == "hero" || self.class.name == "warrior" || self.class.name == "rogue"
     }
 
-    fn new(class: &'static Class, level: i32) -> Self {
+    fn new(class: &'static Class, level: i32, sword: Option<equipment::Sword>,
+           shield: Option<equipment::Shield>) -> Self {
         let mut character = Self {
             class,
-            sword: None,
-            shield: None,
+            class_name: class.name.to_string(),
```

Comment:
you should be able to pull this off without this extra field (which looks redundant with `class.name()`)

it will probably require that you switch from `class: &'static Class` to `class: Class` in the character struct, and making the classes serializable + clonable

Reply:
I was having some issues with adding that functionality into the code earlier, I can give it another go and see how it turns out